sql/src/main/java/io/crate/metadata/doc/DocTableInfoBuilder.java

```diff
@@ -85,12 +86,21 @@ private DocIndexMetaData docIndexMetaData() {
         if (metaData.getTemplates().containsKey(templateName)) {
             docIndexMetaData = buildDocIndexMetaDataFromTemplate(ident.indexName(), templateName);
             createdFromTemplate = true;
+            // We need all concrete indices, regardless of their state, for operations such as reopening.
             concreteIndices = indexNameExpressionResolver.concreteIndexNames(
                 state, IndicesOptions.lenientExpandOpen(), ident.indexName());
+            // We also need all concrete open indices for operations that involve routing, such as SELECT.
+            // This is so closed indices are not included in the routing. This is done by ignoring unavailable indices
+            // and by supplying the indexNameExpressionResolver with the parameter to ignore indices, and an
+            // IndicesOptions that forbids closed indices.
```

Comment:
thinks this is a bit verbose ;) just say: get only open tables, closed tables must not appear at the routing

routing is not only used for SELECT ;)